### PR TITLE
chore: e2e improve docker cache

### PR DESCRIPTION
**Pull request by RafilxTenfen**
- Add multiple CI jobs to run e2e
- `e2e-docker-build-*` builds docker and saves it as an artifact
- `e2e-run-*` runs each e2e test individually
- remove the need of sudo on running e2e tests due to removing error check in ClearResources :smiling_face_with_tear:  
it errors out due to `unlinkat /tmp/bbn-e2e-testnet-2820217771/bbn-test-a/bbn-test-a-node-babylon-default-a-2/ibc_08-wasm/state/wasm: permission denied`  
- removing the need of sudo to run e2e speeds up the run because it uses the go cache created from job `build`

TLDR; It is not a pretty solution, but reduces the CI running time of e2e
> Accepting suggestions/inputs to make it better 

image

:heavy_check_mark: CI time reduced to 8min

### CODE REVIEW

**Review by liam-icheng-lai — APPROVED** [review]
💯

File: contrib/images/babylond/Dockerfile [hunk 1]
```diff
@@ -2,57 +2,38 @@ FROM golang:1.21 AS build-env
...
 # Cosmwasm - Download correct libwasmvm version
-COPY --from=build-env /go/src/github.com/babylonlabs-io/babylon/go.mod /tmp
-RUN WASMVM_VERSION=$(grep github.com/CosmWasm/wasmvm /tmp/go.mod | cut -d' ' -f2) && \
-    wget https://github.com/CosmWasm/wasmvm/releases/download/$WASMVM_VERSION/libwasmvm.$(uname -m).so \
-        -O /lib/libwasmvm.$(uname -m).so && \
-    # verify checksum
-    wget https://github.com/CosmWasm/wasmvm/releases/download/$WASMVM_VERSION/checksums.txt -O /tmp/checksums.txt && \
-    sha256sum /lib/libwasmvm.$(uname -m).so | grep $(cat /tmp/checksums.txt | grep libwasmvm.$(uname -m) | cut -d ' ' -f 1)
-RUN rm -f /tmp/go.mod
+RUN wget https://github.com/CosmWasm/wasmvm/releases/download/$WASMVM_VERSION/libwasmvm.x86_64.so \
```
Comment (KonradStaniec) [inline contrib/images/babylond/Dockerfile:36]: why can 't we leave the old scripts which automatically determines the environment and verify checksum ?

**Review by KonradStaniec — APPROVED** [review]
Just one question, otherwise LGTM! great work!

File: contrib/images/babylond/Dockerfile (hunk 1 above)
Comment (RafilxTenfen) [reply contrib/images/babylond/Dockerfile:36]: we can, I can rollback those changes

Comment (RafilxTenfen) [reply contrib/images/babylond/Dockerfile]: done at https://github.com/babylonlabs-io/babylon/pull/16/commits/0943bc662ab88e407c6bd930f005499dd1335287

File: test/e2e/configurer/current.go [hunk 2]
```diff
@@ -40,12 +40,11 @@ func (cb *CurrentBranchConfigurer) ConfigureChains() error {
 
 func (cb *CurrentBranchConfigurer) ConfigureChain(chainConfig *chain.Config) error {
 	cb.t.Logf("starting e2e infrastructure from current branch for chain-id: %s", chainConfig.Id)
-	tmpDir, err := os.MkdirTemp("", "bbn-e2e-testnet-")
+	tmpDir, err := os.MkdirTemp("", "bbn-e2e-testnet-*")
```
Comment (maurolacy) [inline test/e2e/configurer/current.go:43]: What is this? Just curious.

File: test/e2e/configurer/factory.go [hunk 3]
```diff
@@ -212,11 +218,38 @@ func NewSoftwareUpgradeConfigurer(t *testing.T, isDebugLogEnabled bool, upgradeP
...
+			chain.New(t, containerManager, initialization.ChainAID, updateNodeConfigNameWithIdentifier(validatorConfigsChainA, identifier), nil),
 		},
 		withUpgrade(baseSetup), // base set up with upgrade
 		containerManager,
 		upgradePath,
 		0,
 	), nil
 }
+
+func identifierName(t *testing.T) string {
```
Comment (maurolacy) [inline test/e2e/configurer/factory.go:230]: Why is this needed? Just curious.

Comment (RafilxTenfen) [reply test/e2e/configurer/factory.go:230]: If you would run two or more tests at the same time they might match the name of the container and mess up everything :exploding_head:

File: test/e2e/configurer/current.go (hunk 2 above)
Comment (RafilxTenfen) [reply test/e2e/configurer/current.go:43]:
```go
// If pattern includes a "*", the random string replaces the last "*" instead.
func MkdirTemp(dir, pattern string) (string, error) {
```

Comment (maurolacy) [reply test/e2e/configurer/current.go:43]: Nice.
